src/openrct2/interface/window.c

```diff
@@ -403,6 +403,16 @@ void window_set_window_limit(sint32 value)
  */
 rct_window *window_create(sint32 x, sint32 y, sint32 width, sint32 height, rct_window_event_list *event_handlers, rct_windowclass cls, uint16 flags)
 {
+	//Number must always be kept within valid range.
+	if (gConfigGeneral.window_limit < WINDOW_LIMIT_MIN) {
+		gConfigGeneral.window_limit = WINDOW_LIMIT_MIN;
+		config_save_default();
+	}
+	if (gConfigGeneral.window_limit > WINDOW_LIMIT_MAX) {
+		gConfigGeneral.window_limit = WINDOW_LIMIT_MAX;
+		config_save_default();
+	}
```

Comment:
This would better be moved to the initialisation state, where it loads the config.